errorcodes/errorcodes.go

```diff
@@ -0,0 +1,241 @@
+/*
```

Comment:
Consider changing the package and directory names to be `errors` to make things cleaner. Instances of the string `Error` can then be removed from names and signatures since it is implied by the package name.